go/services/reports/pkg/incident_correlation/incident_merger.go

```diff
@@ -0,0 +1,253 @@
+        %s`, content),
+		},
+	}
+}
+
+type IncidentMergeGroup struct {
+	Title          string           `json:"title"`
+	Summary        string           `json:"summary"`
+	Recommendation string           `json:"recommendation"`
+	Category       string           `json:"category"`
+	Urgency        insights.Urgency `json:"urgency"`
+	IncidentIds    []string         `json:"incidentIds"`
+}
+
+type incidentMergerResponseDto struct {
+	IncidentMergeGroups []IncidentMergeGroup `json:"incidentMergeGroup"`
+}
+
+// Accepts a map of incidents groups, where a key is a group identifier
+func (m *OpenAiIncidentMerger) ScheduleIncidentsMerge(incidentGroups map[string][]repositories.Incident) ([]*repositories.ScheduledIncidentMergerJob, error) {
+
+	completionRequests := make(map[string]*openai.CompletionRequest, len(incidentGroups))
+	if len(incidentGroups) == 0 {
+		return make([]*repositories.ScheduledIncidentMergerJob, 0), nil
+	}
+
+	for groupId, report := range incidentGroups {
+		criterias := array.Map(m.getIncidentMergeCriteria)(report)
+		encodedCriterias, err := m.encodeMergeCriteria(criterias)
+
+		encodedCriteriasPerPacket := splitting.SplitStringsIntoPackets(encodedCriterias, m.client.ContextSizeBytes)
+
+		if err != nil {
+			return nil, err
+		}
+		for idx, packet := range encodedCriteriasPerPacket {
+			messages := m.createSummaryRequestMessage(packet)
+
+			completionRequests[getGroupPacketId(groupId, idx)] = &openai.CompletionRequest{
+				Messages:       messages,
+				Temperature:    m.client.Temperature,
+				ResponseFormat: openai.CreateJsonReponseFormat("incidentMerger", incidentMergerResponseDto{}),
+				Model:          m.client.Model(),
+			}
+		}
+	}
+
+	completionReuqestsPerBatch, err := m.client.SplitCompletionReqestsByBatchSize(completionRequests)
+	if err != nil {
+		m.logger.Error("Failed to split merger requests by batch", zap.Error(err))
+		return nil, err
+	}
+
+	jobs := make([]*repositories.ScheduledIncidentMergerJob, 0, len(completionReuqestsPerBatch))
+	for _, batch := range completionReuqestsPerBatch {
+
+		jobId, err := m.scheduledJobsRepository.InsertScheduledJob(context.Background(), &openai.OpenAiJob{
+			ScheduledAt:        time.Now().UnixMilli(),
+			CompletionRequests: batch,
+			Status:             openai.OpenAiJobStatus__Enqueued,
+		})
+
+		if err != nil {
+			m.logger.Error("Failed to insert scheduled jobs", zap.Error(err))
+			return nil, err
+		}
+
+		jobs = append(jobs, &repositories.ScheduledIncidentMergerJob{
+			Id: jobId,
+		})
+
+	}
+
+	return jobs, nil
+}
+
+func getGroupPacketId(groupId string, packetId int) string {
+	return fmt.Sprintf("%s-%d", groupId, packetId)
+}
+
+func getGroupFromEncodedGroupPacketId(groupPacketId string) string {
+	return strings.Split(groupPacketId, "-")[0]
+}
+
+func (m *OpenAiIncidentMerger) TryGettingIncidentMergerJobIfFinished(job *repositories.ScheduledIncidentMergerJob) (map[string][]IncidentMergeGroup, error) {
```

Comment:
What "Try" indicates here?

Reply:
Try is mentioned to focus on the part that this funcion assumes that the job is ready, if the job is not completed  it will return an error instead